packages/webpack/lib/utils/loader.js

```diff
@@ -1,23 +1,26 @@
 const { getOptions } = require('loader-utils');
 
 const getMixins = (target, config) =>
-  config.mixins[target].map(
-    (mixin) => `(function(m) { return m.default || m })(require('${mixin}'))`
-  );
+  `[${config.mixins[target]
+    .map(
+      (mixin) => `(function(m) { return m.default || m })(require('${mixin}'))`
+    )
+    .join(',')}]`;
 
 const getConfig = (target, config) =>
-  JSON.stringify(config)
-    .replace(
-      new RegExp(`"${config.rootDir}(.*?)"`, 'g'),
-      target === 'server' ? 'expand(".$1")' : '".$1"'
-    )
-    .replace(
-      /"mixins":{.*?}/,
-      `"mixins":[${getMixins(target, config).join(', ')}]`
-    );
+  JSON.stringify(
+    Object.assign({}, config._config, { mixins: undefined })
+  ).replace(
+    new RegExp(`"${config.rootDir}(.*?)"`, 'g'),
+    target === 'server' ? 'expand(".$1")' : '".$1"'
+  );
+
+const getConfigAndMixins = (...args) =>
+  `{ config: ${getConfig(...args)}, mixins: ${getMixins(...args)} }`;
 
-const getModule = (target, config) =>
-  (target === 'server'
+const getModule = (target, config) => {
+  const configAndMixins = getConfigAndMixins(target, config);
```

Comment:
We talked about allowing to configure the loader without mixins - how about adding another option to the loader, for example: `noMixins: true` or something similar, that then uses `getConfig(target, config)` instead of `getConfigAndMixins()`?
That way the resulting bundle won't include all the code required for the mixins.

Reply:
As we discussed IRL: I will make sure the config loader does not explode if the target is non-existing. Thus, if we consider the loader with a non-existing or even `undefined` target, that will work as you describe.